nixos/lib/test-driver/src/test_driver/machine/ocr.py

```diff
@@ -0,0 +1,121 @@
+import multiprocessing
+import os
+import shutil
+import subprocess
+
+from test_driver.errors import MachineError
+
+
+def perform_ocr_on_screenshot(screenshot_path: str) -> str:
+    """
+    Perform OCR on a screenshot that contains text.
+    Returns a string with all words that could be found.
+    """
+    variants = perform_ocr_variants_on_screenshot(screenshot_path, False)[0]
+    if len(variants) != 1:
+        raise MachineError(f"Received wrong number of OCR results: {len(variants)}")
+    return variants[0]
+
+
+def perform_ocr_variants_on_screenshot(
+    screenshot_path: str, variants: bool = True
+) -> list[str]:
+    """
+    Same as perform_ocr_on_screenshot but will create variants of the images
+    that can lead to more words being detected.
+    Returns a string with words for each variant.
+    """
+    if shutil.which("tesseract") is None:
+        raise MachineError("OCR requested but `tesseract` is not available")
+
+    # Tesseract runs parallel on up to 4 cores.
+    # Docs suggest to run it with OMP_THREAD_LIMIT=1 for hundreds of parallel
+    # runs. Our average test run is somewhere inbetween.
+    # https://github.com/tesseract-ocr/tesseract/issues/3109
+    processes = int(os.process_cpu_count() / 4)
+    with multiprocessing.Pool(processes=processes) as pool:
+        image_paths: list[str] = [screenshot_path]
+        if variants:
+            image_paths.extend(
+                pool.starmap(
+                    _preprocess_screenshot,
+                    [(screenshot_path, False), (screenshot_path, True)],
+                )
+            )
+        return pool.map(_run_tesseract, image_paths)
+
+
+def _run_tesseract(image: str) -> str:
+    # tesseract --help-oem
+    # OCR Engine modes (OEM):
+    #  0|tesseract_only          Legacy engine only.
+    #  1|lstm_only               Neural nets LSTM engine only.
+    #  2|tesseract_lstm_combined Legacy + LSTM engines.
+    #  3|default                 Default, based on what is available.
+    ocr_engine_mode = 2
```

Comment:
Wound using an Enum make this clearer ?

Outside of this function:
```python

class OCREngineMode(Enum):
    """
    Enum representing the OCR Engine modes for Tesseract.
    """
    TESSERACT_ONLY = 0
    # Legacy engine only.
    LSTM_ONLY = 1
    # Neural nets LSTM engine only.
    TESSERACT_LSTM_COMBINED = 2
    # Legacy + LSTM engines.
    DEFAULT = 3
    # Default, based on what is available.
```

```suggestion
    ocr_engine_mode = OCREngineMode.TESSERACT_LSTM_COMBINED
```